CMakeLists.txt

```diff
@@ -154,30 +157,101 @@ else()
     message(STATUS "IPO / LTO not supported: <${error}>")
 endif()
 
-# auto optimize
-if (AUTO_OPTIMIZE)
-    include(OptimizeForArchitecture)
-    AutodetectHostArchitecture()
-    OptimizeForArchitecture()
-    # remove global definition for portable build
-    foreach(_flag ${_enable_vector_unit_list})
-        string(TOUPPER "${_flag}" _flag)
-        string(REPLACE "." "_" _flag "__${_flag}__")
-        remove_definitions("-D${_flag}")
-    endforeach(_flag)
+CHECK_CXX_COMPILER_FLAG("-march=native" COMPILER_SUPPORTS_MARCH_NATIVE)
+if(COMPILER_SUPPORTS_MARCH_NATIVE)
+    set(MARCH_NATIVE_FLAGS "${MARCH_NATIVE_FLAGS} -march=native -mtune=native")
+    set(MARCH_NATIVE_OPT ON)
 endif()
 
-# Compiler specific environments
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  set(CLANG 1)
+CHECK_CXX_COMPILER_FLAG("-mavx" HAVE_AVX)
+if(HAVE_AVX)
+    set(AVX_COMPILE_FLAGS "-mavx")
+    add_definitions("-DHAVE_AVX")
+endif(HAVE_AVX)
+
+CHECK_CXX_COMPILER_FLAG("-mavx2" HAVE_AVX2)
+if(HAVE_AVX2)
+    set(AVX2_COMPILE_FLAGS "-mavx2")
+    add_definitions("-DHAVE_AVX")
 endif()
 
-CHECK_CXX_COMPILER_FLAG("-mfpu=neon" HAVE_NEON)
-if(HAVE_NEON)
-    set(NEON_COMPILE_FLAGS "-mfpu=neon")
-    add_definitions("-DHAVE_NEON")
+CHECK_CXX_COMPILER_FLAG("-mfma" HAVE_FMA)
+if(HAVE_FMA)
+    set(FMA_COMPILE_FLAGS "-mfma")
+    add_definitions("-DHAVE_FMA")
+endif()
+
+CHECK_CXX_COMPILER_FLAG("-msse4.1" HAVE_SSE4_1)
+if(HAVE_SSE4_1)
+    set(SSE4_1_COMPILE_FLAGS "-msse4.1")
+    add_definitions("-DHAVE_SSE4_1")
 endif()
 
+if(NOT APPLE)
```

Comment:
NEON is only available for ARM. As Apple is going to replace Intel by ARM, this would be wrong soon. We should check for ARM instead of checking for `NOT APPLE`.

Reply:
Likewise the checks for Intel features (AVX, ...) could be run only for Intel architecture.

Reply:
to make it short: see [log from travis](https://travis-ci.org/github/tesseract-ocr/tesseract/jobs/703646220) - neon is tested positively in apple build, but our code for neon is not expecting it.
IMO iphones and ipad are already ARM devices, so we can remove this restriction when tesseract support for neon will be adjusted to this.